qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/template/datamodel/TemplateGlobalAnnotationSupport.java

```diff
@@ -0,0 +1,154 @@
+import com.redhat.qute.jdt.internal.resolver.ITypeResolver;
+import com.redhat.qute.jdt.template.datamodel.AbstractAnnotationTypeReferenceDataModelProvider;
+import com.redhat.qute.jdt.template.datamodel.SearchContext;
+import com.redhat.qute.jdt.utils.AnnotationUtils;
+
+/**
+ * @TemplateGlobal annotation support.
+ *
+ *                 <code>
+ * &#64;TemplateGlobal
+ * public class Globals {
+ *
+ * 		static int age = 40;
+ *
+ * 		static Color[] myColors() {
+ * 			return new Color[] { Color.RED, Color.BLUE };
+ * 		}
+ *
+ * 		&#64;TemplateGlobal(name = "currentUser")
+ * 		static String user() {
+ * 			return "Mia";
+ * 		}
+ * }
+ * </code>
+ *
+ *
+ * @see https://quarkus.io/guides/qute-reference#global_variables
+ *
+ */
+public class TemplateGlobalAnnotationSupport extends AbstractAnnotationTypeReferenceDataModelProvider {
+
+	private static final Logger LOGGER = Logger.getLogger(TemplateGlobalAnnotationSupport.class.getName());
+
+	private static final String[] ANNOTATION_NAMES = {
+		TEMPLATE_GLOBAL_ANNOTATION
+	};
+
+	@Override
+	protected String[] getAnnotationNames() {
+		return ANNOTATION_NAMES;
+	}
+
+	@Override
+	protected void processAnnotation(IJavaElement javaElement, IAnnotation annotation, String annotationName,
+		SearchContext context, IProgressMonitor monitor) throws JavaModelException {
+		if (javaElement.getElementType() != IJavaElement.TYPE) {
+			return;
+		}
+		IType type = (IType) javaElement;
+		String name = null;
+		boolean hasTemplateGlobal = false;
+		for (IAnnotation typeAnnotation : type.getAnnotations()) {
+			if (AnnotationUtils.isMatchAnnotation(typeAnnotation, TEMPLATE_GLOBAL_ANNOTATION)) {
+				hasTemplateGlobal = true;
+				name = AnnotationUtils.getAnnotationMemberValue(typeAnnotation, TEMPLATE_GLOBAL_ANNOTATION_NAME);
+				if (StringUtils.isNotEmpty(name)) {
+					break;
+				}
+			}
+		}
+		if (!hasTemplateGlobal) {
+			return;
+		}
+
+		ITypeResolver typeResolver = QuteSupportForTemplate.createTypeResolver(type);
+
+		// Loop for static fields
+		IField[] fields = type.getFields();
+		for (IField field : fields) {
+			collectResolversForTemplateGlobal(field, name, context.getDataModelProject().getValueResolvers(),
+				typeResolver, monitor);
+		}
+
+		// Loop for static methods
+		IMethod[] methods = type.getMethods();
+		// Method-level annotations override the class-level annotation
+		for (IMethod method : methods) {
+			collectResolversForTemplateGlobal(method, name, context.getDataModelProject().getValueResolvers(),
+				typeResolver, monitor);
+		}
+	}
+
+	private void collectResolversForTemplateGlobal(IMember member, String name, List<ValueResolverInfo> resolvers,
+		ITypeResolver typeResolver, IProgressMonitor monitor) {
+		try {
+			// every non-void non-private static method that declares no parameters and
+			// every non-private static field is considered a global variable
+			if (!Modifier.isPrivate(member.getFlags()) && Modifier.isStatic(member.getFlags())) {

```

Comment:
It is not enough, I suggest that you read Javadoc at https://github.com/quarkusio/quarkus/blob/5e6fe630375c96b18228f54c6537f445f3ee1743/independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateGlobal.java#L10 which explains rules that field and method must match (ex : static, etc) and implement those rules.

Reply:
So both a field and method must be non private and static, and a method must have no parameters and return void, correct?

Reply:
Yes, it misses the conditions for methods (no parameters and return void)